db/controller/topics.controller.js

```diff
@@ -51,4 +56,48 @@ exports.getTopics = async (req, res) => {
     } catch (err) {
       next(err);
     }
-  };
\ No newline at end of file
+  };
+
+
+  exports.postComment = async (req, res, next) => {
+    try {
+      const { article_id } = req.params
+      const { username, body } = req.body
+      insertCommentByArticleId(article_id, username, body)
+          .then((comment) => {
+              res.status(201).send({ comment })
+          })
+          .catch((err) => {
+              next(err)
+          })
+    } catch (err) {
+        if (err.code === "23503") {
+            res.status(404).send({ msg: "User not found" });
+        } else if (err.code === "22P02") {
+            res.status(400).send({ msg: "Invalid article_id format" });
+        } else {
+            next(err);
+        }
+    }
+};
+
+  exports.postComment = async (req, res, next) => {
+   try {
+    const { article_id } = req.params
+    const { username, body } = req.body
+    if (!username) {
+      return res.status(400).send({ message: "Username is required" });
+  }
+    insertCommentByArticleId(article_id, username, body)
+        .then((comment) => {
+            res.status(201).send({ comment })
+        })
+        .catch((err) => {
+            next(err)
+        })
+   } catch(err) {
+    console.log(err)
+   }
+};
```

Comment:
This (same as the above) is mixing async await and promises together. I would recommend sticking to one for consistency and readability. 

This is also resulting in two catch blocks which is not ideal either.

Promise notes:
https://l2c.northcoders.com/courses/sd-notes/fundamentals#sectionId=promises,step=intro

Async await notes:
https://l2c.northcoders.com/courses/sd-notes/fundamentals#sectionId=async-await,step=intro